source/base/Allocator.ooc

```diff
@@ -0,0 +1,25 @@
+/* This file is part of magic-sdk, an sdk for the open source programming language magic.
+ *
+ * Copyright (C) 2016 magic-lang
+ *
+ * This software may be modified and distributed under the terms
+ * of the MIT license.  See the LICENSE file for details.
+ */
+
+import Owner
+
+AbstractAllocator: abstract class {
```

Comment:
You can also use template classes to create composable allocators (I'm thinking about Andrei Alexandrescu's design of the D std.allocator), although they are quite broken right now.  

By the way, another nice way to expose allocators to any ooc code would be to add a global `current_allocator` variable and use it to allocate and free (in `Object alloc` and `Object free`) although that means that you then have to keep track of what variables were allocated with what allocator etc.  

With such a design, I expect you would use a single well tuned allocator for most allocations and switch allocators in specific places (for example, `current_allocator = SomePoolAllocator; /* allocate thousands of small objects here, use them, free them */; current_allocator = SystemWideAllocator`).